x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizationEngine.java

```diff
@@ -667,6 +697,103 @@ public IndicesAccessControl getIndicesAccessControl() {
         }
     }
 
+    /**
+     * Holds information about authorization of a parent action which is used to pre-authorize its child actions.
+     *
+     *  @param action the parent action
+     */
+    record ParentActionAuthorization(String action) implements Writeable, ToXContent {
+
+        public static final String THREAD_CONTEXT_KEY = "_xpack_security_authorization";
+        private static final ParseField ACTION = new ParseField("action");
+
+        private static final ConstructingObjectParser<ParentActionAuthorization, String> PARSER = new ConstructingObjectParser<>(
+            "authorization",
+            true,
+            (a) -> new ParentActionAuthorization((String) a[0])
+        );
+        static {
+            PARSER.declareString(constructorArg(), ACTION);
+        }
```

Comment:
nit: It looks like the `ParentActionAuthorization` record is never passed around in "XContent" form (such as request or response bodies).
I would avoid adding code that's only exercised in tests, so I would suggest you remove the parser.

Reply:
Agreed. I've removed the XContent logic. https://github.com/elastic/elasticsearch/pull/91886/commits/7b5e28f9c94535c4355ce297acae73a526304ff4